icu4j/main/classes/core/src/com/ibm/icu/util/ULocale.java

```diff
@@ -1239,147 +1563,54 @@ public static String canonicalize(String localeID){
-                    .append(lscvToID(replaceLocale.getLanguage(),
-                        parser.getScript().isEmpty() ? replaceLocale.getScript() : parser.getScript() ,
-                        parser.getCountry().isEmpty() ? replaceLocale.getCountry() : parser.getCountry() ,
-                        parser.getVariant()))
-                    .append(parser.getName().substring(parser.getBaseName().length()))
-                    .toString());
-            } catch (MissingResourceException e) {
-            }
-        } catch (MissingResourceException e) {
         }
 
-        // If the BCP 47 region subtag matches the type attribute of a
-        // territoryAlias element in Supplemental Data, replace the language
-        // subtag with the replacement value, as follows:
-        if (!parser.getCountry().isEmpty()) {
-            try {
-                String replacements[] = UResourceBundle.getBundleInstance(ICUData.ICU_BASE_NAME,
-                    "metadata", ICUResourceBundle.ICU_DATA_CLASS_LOADER)
-                    .get("alias")
-                    .get("territory")
-                    .get(parser.getCountry())
-                    .get("replacement")
-                    .getString()
-                    .split(" ");
-                String replacement = replacements[0];
-                // If there is a single territory in the replacement, use it.
-                // If there are multiple territories:
-                // Look up the most likely territory for the base language code (and script, if there is one).
-                // If that likely territory is in the list, use it.
-                // Otherwise, use the first territory in the list.
-                if (replacements.length > 1) {
-                    String likelyCountry = ULocale.addLikelySubtags(
-                        new ULocale(lscvToID(parser.getLanguage(), parser.getScript(), null, parser.getVariant())))
-                        .getCountry();
-                    for (String country : replacements) {
-                        if (country.equals(likelyCountry)) {
-                            replacement = likelyCountry;
-                            break;
-                        }
-                    }
-                }
-                parser = new LocaleIDParser(
-                    (new StringBuilder(localeID.length()))
-                    .append(lscvToID(parser.getLanguage(), parser.getScript(), replacement, parser.getVariant()))
-                    .append(parser.getName().substring(parser.getBaseName().length()))
-                    .toString());
-            } catch (MissingResourceException e) {
-            }
+        return parser.getName();
+    }
+
+    private static boolean isKnownCanonicalizedLocale(String name) {
+        if (name.equals("c") || name.equals("en") || name.equals("en_US")) {
+            return true;
         }
+        if (gKnownCanonicalizedCases == null) {
```

Comment:
Remember to declare the function "synchronized", or synchronize(getClass()) a block surrounding this lazy init.